### PR TITLE
remove host name if sharing the same one during uid resolving

### PR DESCRIPTION
**Pull request by live1206**
#3885

### CODE REVIEW

File: docs/specs/xref.yml [hunk 1]
```diff
@@ -20,8 +20,10 @@ outputs:
     ["info","at-uid-not-found","Cannot find uid 'a' using xref '@a'","docs/a.md"]
 ---
 # Restore xref map from configured url and use the map for resolving
+# hostname should be removed if docset sharing the same one
 inputs:
   docfx.yml: |
+    baseUrl: docs.microsoft.com
```
Comment (yufeih) [inline docs/specs/xref.yml]: `baseUrl` contains `https://`, so this should be 'https://docs.microsoft.com'

Comment (live1206) [reply docs/specs/xref.yml]: ok

File: src/docfx/build/xref/XrefMap.cs [hunk 2]
```diff
@@ -70,6 +71,16 @@ internal class XrefMap
             // xrefSpec.displayPropertyName -> xrefSpec.name -> uid
             string display = !string.IsNullOrEmpty(displayPropertyValue) ? displayPropertyValue : (!string.IsNullOrEmpty(name) ? name : uid);
             return (null, resolvedHref, display, referencedFile);
+
+            string RemoveHostnameIfSharingTheSameOne(string input)
+            {
+                var hostname = rootFile.Docset.Config.BaseUrl;
+                if (input.StartsWith(hostname, StringComparison.OrdinalIgnoreCase))
+                {
+                    return input.Substring(input.IndexOf(hostname) + hostname.Length);
```
Comment (yufeih) [inline src/docfx/build/xref/XrefMap.cs]:
```suggestion
                    return input.Substring(hostname.Length);
```

File: src/docfx/build/xref/XrefMap.cs [hunk 3]
```diff
@@ -70,6 +71,16 @@ internal class XrefMap
             // xrefSpec.displayPropertyName -> xrefSpec.name -> uid
             string display = !string.IsNullOrEmpty(displayPropertyValue) ? displayPropertyValue : (!string.IsNullOrEmpty(name) ? name : uid);
             return (null, resolvedHref, display, referencedFile);
+
+            string RemoveHostnameIfSharingTheSameOne(string input)
```
Comment (OsmondJiang) [inline src/docfx/build/xref/XrefMap.cs]: do we need consider the http or https case? same host name but different schema

Comment (live1206) [reply src/docfx/build/xref/XrefMap.cs]: since the hostname contains `http(s)`, it should match?

Comment (yufeih) [reply src/docfx/build/xref/XrefMap.cs]: That'll make things a lot complicated, url comparison is itself very complex if you also consider port, www, etc. So we just right now use simple string comparison.

**Pull request merged**